docs/proposals/formatted-output.md

```diff
@@ -0,0 +1,348 @@
+
+- Why ORAS chooses [Go template](https://pkg.go.dev/text/template)?
+
+Go template is a powerful method to customize output you want It allows users to manipulate the output format of certain commands. It provides access to data objects and additional functions that are passed into the template engine programmatically. It also has some useful libraries that have strong functions for Go’s template language to manipulate the output data, such as [Sprig](https://masterminds.github.io/sprig/).
+
+- What's the difference of the output when use `--output` and `--format` in `oras manifest fetch`?
+
+`--output` can generate a file with raw JSON data of the image manifest. `format` can display prettified JSON in the standard out.
+
+```bash
+oras manifest fetch $REGISTRY/$REPO:$TAG --output manifest.json
+```
+
+See the content in the `manifest.json`. It should be raw JSON data of the fetched image manifest.
+
+```json
+{"schemaVersion":2,"mediaType":"application/vnd.oci.image.manifest.v1+json","artifactType":"application/vnd.unknown.artifact.v1","config":{"mediaType":"application/vnd.oci.empty.v1+json","digest":"sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a","size":2,"data":"e30="},"layers":[{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65","size":14477,"annotations":{"org.opencontainers.image.title":"sbom.spdx"}},{"mediaType":"application/vnd.oci.image.layer.v1.tar","digest":"sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69","size":6639,"annotations":{"org.opencontainers.image.title":"scan-test-verify-image.json"}}],"annotations":{"org.opencontainers.image.created":"2023-12-13T15:08:49Z"}}
+```
+
+See the prettified JSON output of an image manifest when use `--format json`:
+
+```bash
+oras manifest fetch $REGISTRY/$REPO:$TAG --format json
+```
+
+```json
+{
+  "schemaVersion": 2,
+  "mediaType": "application/vnd.oci.image.manifest.v1+json",
+  "artifactType": "application/vnd.unknown.artifact.v1",
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2,
+    "data": "e30="
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar",
+      "digest": "sha256:6cb759c4296e67e35b0367f3c0f51dfdb776a0c99a45f39d0476e43d82696d65",
+      "size": 14477,
+      "annotations": {
+        "org.opencontainers.image.title": "sbom.spdx"
+      }
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar",
+      "digest": "sha256:54c0e84503c8790e03afe34bfc05a5ce45c933430cfd9c5f8a99d2c89f1f1b69",
+      "size": 6639,
+      "annotations": {
+        "org.opencontainers.image.title": "scan-test-verify-image.json"
+      }
+    }
+  ],
+  "annotations": {
+    "org.opencontainers.image.created": "2023-12-13T15:08:49Z"
+  }
+}
+```
```

Comment:
I think can follow up with with a similar output for oras copy. I would like find out all the referrers that did get copied and their digest and artifactTypes that got copied over. 

The console output cannot be used to track everything that got copied.

Reply:
@sajayantony 
I am still a bit confusing about using `--format` in `oras copy`. `oras copy` will not change the metadata of the copied image manifest and its referrers. 

Should we use [oras discover](https://github.com/oras-project/oras/pull/1199/files#diff-ddd85908fb8fee8c57a3726cb042b0572e9b4b333000356816268399ced78cccR275) with `--format` to filter out the `digest` and `artifactTypes` of referrers on a source artifact instead?

Reply:
I was suggesting we create another issue since the usecase is not clear and we can follow up as an other PR if necessary.

Reply:
@sajayantony Sounds good. I will create an issue to track it, in case any users have specific scenarios or feature requests on it in the future.

Reply:
@sajayantony Created an issue to track potential formatted JSON output support for oras copy: https://github.com/oras-project/oras/issues/1226